### PR TITLE
start cwd builtin

### PR DESCRIPTION
**Pull request by ekelen**
basic chdir working
add str util
makefile update

Expected behavior:
```sh
ftp> pwd // 257 "~/ft_p" is your current location
ftp> cd srcs
ftp> pwd // 257 "~/ft_p/srcs" is your current location
ftp> cd ../.. // 550 You are not authorized.
```

### CODE REVIEW

File: srcs/commons/utils_str.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+/*   By: ekelen <erskelen@gmail.com>                +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2019/09/07 15:15:49 by ekelen            #+#    #+#             */
+/*   Updated: 2019/09/07 22:27:00 by ekelen           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../inc/ft_p.h"
+
+size_t		ft_count_strs(char **s)
```
Comment (Failte21) [inline srcs/commons/utils_str.c]: I don't know if relevant but why not using a i? (while s[i]) so we don't mess with the pointer?

File: srcs/commons/utils_str.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+size_t	ft_count_words(char const *s, char c)
+{
+	size_t nwords;
+	size_t is_start;
+
+	nwords = 0;
+	is_start = 0;
+	while (*s)
```
Comment (Failte21) [inline srcs/commons/utils_str.c]: same here

File: srcs/server/process_commands_2.c [hunk 3]
```diff
@@ -1,6 +1,16 @@
-#include "../../inc/ftp_server.h"
+/* ************************************************************************** */
```
Comment (Failte21) [inline srcs/server/process_commands_2.c]: would rather not have headers until the end (to make sure we don't have each other headers on the vogsphere repo)

File: srcs/server/process_commands_2.c [hunk 4]
```diff
@@ -20,4 +30,46 @@ void    handle_pwd(t_server_handler *handler, char **args)
...
+    {
+        send_reply(handler, REP_ACTION_NOT_TAKEN);
+        ft_strdel(&startdir);
+        return;
+    }
+    // get absolute directory after move
+    dirname = getcwd(dirname, 0);
+    // if it's higher than user root, move back
+    if (ft_strncmp(dirname, handler->root, ft_strlen(handler->root))) {
+        send_reply(handler, "550 You are not authorized.\r\n");
```
Comment (Failte21) [inline srcs/server/process_commands_2.c]: you forgot a define for this reply

File: srcs/server/server_handler.c [hunk 5]
```diff
@@ -7,5 +7,11 @@ t_server_handler	*init_server_handler()
 	handler = (t_server_handler *)malloc(sizeof(t_server_handler));
 	if (handler == NULL)
 		exit_error("Malloc error allocating memory for server_handler");
+	handler->root = NULL;
+
+	if ((handler->root = getcwd(handler->root, 0)) == NULL)
+		exit_error("Malloc error allocating for root dirname");
+	ft_putstr_fd("(debug) set handler->root", 2);
```
Comment (Failte21) [inline srcs/server/server_handler.c]: Do we want to keep those logs?

File: srcs/commons/utils_str.c (hunk 1 above)
Comment (ekelen) [reply srcs/commons/utils_str.c]: agree, changing

File: srcs/commons/utils_str.c (hunk 2 above)
Comment (ekelen) [reply srcs/commons/utils_str.c]: agree

File: srcs/server/process_commands_2.c (hunk 4 above)
Comment (ekelen) [reply srcs/server/process_commands_2.c]: I made it up, I'll add a todo

File: srcs/server/server_handler.c (hunk 5 above)
Comment (ekelen) [reply srcs/server/server_handler.c]: nope, artefact of bad rebase

**Comment by Failte21** [comment]
🦁 ✅

**Pull request merged**